src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs

```diff
@@ -3469,6 +3506,49 @@ static void Main(string[] args)
         }
 
         [Fact]
+        public void ForEachVariable_Update_Leaf2()
+        {
+            string src1 = @"
+class Test
+{
+    public static (int, bool)[] e1 = new (int, bool)[1];
+    public static (int, bool)[] e2 = new (int, bool)[1];
+    
+    static void Main(string[] args)
+    {
+        <AS:0>System.Console.Write();</AS:0>
+    }
+}";
+            string src2 = @"
+class Test
+{
+    public static (int, bool)[] e1 = new (int, bool)[1];
+    public static (int, bool)[] e2 = new (int, bool)[1];
+    
+    static void Main(string[] args)
+    {
+        foreach ((int b1, bool b2) in e1)
+        {
+            foreach (var c in e1)
+            {
+                foreach ((var a1, var a2) in e1)
+                {
+                    <AS:0>System.Console.Write();</AS:0>
+                }
+            }
+        }
+    }
+}";
+            var edits = GetTopEdits(src1, src2);
+            var active = GetActiveStatements(src1, src2);
+
+            edits.VerifyRudeDiagnostics(active,
+                Diagnostic(RudeEditKind.InsertAroundActiveStatement, "foreach (var c in e1)", "foreach statement"),
+                Diagnostic(RudeEditKind.InsertAroundActiveStatement, "(int b1, bool b2)", "deconstruction"),
+                Diagnostic(RudeEditKind.InsertAroundActiveStatement, "(var a1, var a2)", "deconstruction"));
+        }
+
+        [Fact]
```

Comment:
This was fixed in the previous iteration. Thanks!

---
In reply to: [114644807](https://github.com/dotnet/roslyn/pull/19114#discussion_r114644807) [](ancestors = 114644807)